x/ecocredit/msgs.go

```diff
@@ -60,10 +67,17 @@ func (m *MsgSendRequest) ValidateBasic() error {
 			return err
 		}
 
-		_, err = math.ParseNonNegativeDecimal(iss.RetiredUnits)
+		retiredUnits, err := math.ParseNonNegativeDecimal(iss.RetiredUnits)
 		if err != nil {
 			return err
 		}
+
+		if !retiredUnits.IsZero() {
+			err = validateLocation(iss.RetirementLocation)
```

Comment:
@blushi @clevinson do we always require a valid location? Could someone retire credits without providing a location theoretically?

Reply:
I don't think so, this should be required as long as credits get retired

Reply:
@blushi is it worth a final check with Ron?